File: teste.py
# exemplo_cliente_py.py
from cliente_lib import encurta, resolve, remove_url

# buffers para receber resultados (tamanhos máximos)
codigo = bytearray(20)
url = bytearray(500)

# Encurtar
if encurta("https://www.exemplo.com/muito/longa", codigo) == 0:
    print(f"Encurtado: {codigo.decode().strip('\0')}")
else:
    print("Falha ao encurtar")

# Resolver
if resolve(codigo.decode().strip('\0'), url) == 0:
    print(f"URL original: {url.decode().strip('\0')}")
else:
    print("Falha ao resolver")

# Remover
if remove_url(codigo.decode().strip('\0')) == 0:
    print("Removido com sucesso")
else:
    print("Falha ao remover")

File: teste.c
#include <stdio.h>
#include "cliente.h"

int main() {
    char codigo[1024];
    char url[1024];

    if (encurta("https://www.google.com", codigo) == 0) {
        printf("Encurtado: %s\n", codigo);
    } else {
        printf("Erro ao encurtar URL\n");
        return 1;
    }
    if (resolve(codigo, url) == 0) {
        printf("URL original: %s\n", url);
    } else {
        printf("Erro ao resolver URL\n");
        return 1;
    }
    if (remove_url(codigo) == 0) {
        printf("Removido com sucesso!\n");
    } else {
        printf("Erro ao remover URL\n");
        return 1;
    }

    return 0;
}
